### ISSUE TITLE
Don't collapse folder when clicking on folder name

### ISSUE THREAD

**Issue by adamreisnz** [atom/tree-view#1222]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
I would like to propose that clicking on a folder name in tree view does *not* collapse that folder, unless explicitly clicked on the caret area next to it.

The reason for this is twofold.

1) This is non-standard behaviour and inconsistent with how OS's generally work. OS X for example does not open or collapse. One has to click on the caret in order to open/collapse folders:

![ex1](https://user-images.githubusercontent.com/490562/35894552-27bb9c0e-0c18-11e8-9404-36ceb8fe276e.gif)

2) This is frustrating when you are using keyboard shortcuts to create new files. In that case, I might want to create a new file in an open folder, so I go to tree view and left-click on the folder name and press `a`. This creates the file, but due to me clicking on the folder, it has now closed that folder, forcing me to click again to re-open it if I want to see my files again.

I understand why this (probably) has been implemented this way, as it increases the clickable area for opening/collapsing folders, and I like it for opening folders, but I don't think folders should collapse when clicking on the folder name.

### Steps to Reproduce

1. Open/expand a folder in tree view
2. Switch focus to editor
3. Click on a folder name to select it

**Expected behavior:** [What you expect to happen]
Folder stays open and becomes selected.

**Actual behavior:** [What actually happens]
Folder collapses and becomes selected.

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

### Versions
Latest, n/a

**Comment by savetheclocktower** [comment]
Suggestions for possible behavior:

* Only toggle when clicking on the disclosure triangle.
* Always _expand_ a collapsed folder if any part of it is clicked on, but only _collapse_ an expanded folder if the disclosure triangle is clicked on.
* Toggle if any part of the entry is clicked, but _only_ if focus was already inside the tree view pane.
* Only toggle if the entry is double-clicked.

**Comment by adamreisnz** [comment]
I quite like the 3rd suggestion actually, as it might also prevent you from opening/closing folders inadvertently when you just want to focus on the tree view pane.

However, option 4, double clicking to open/close would probably be the easiest to implement and would work great as well. OS X opens the folder when double clicking the name, so the behaviour would be fairly similar 👍

**Comment by ebarault** [comment]
https://atom.io/packages/dbclick-tree-view

**Comment by adamreisnz** [comment]
Thanks, does that still preview the file on single click when double click is enabled?

**Comment by corourke** [comment]
Yes, https://atom.io/packages/dbclick-tree-view has a preference to keep the single click on files. Works great.

**Comment by bigminigame** [comment]
Thanks for this plugin, but this should really be a default option to have, and it should sit right next to the "Allow pending pane item" option. We shouldn't have to code/install a plugin for such a basic ergonomic choice.

**Comment by calebmeyer** [comment]
I'm not opposed to an option, but this click behavior is pretty standard for text editors (at the least, sublime and vs code do this as well). The OS level file browser is different, because clicking on a folder often opens the folder in a right panel of some sort. It doesn't when you specifically have the finder in tree view mode, but I'd argue that's a bug in the finder (which is already the worst GUI file manager of any OS I've tried). In a text editor, we're doing the same thing (showing the files in the folder), but we don't have an extra pane to do it in.

Re: 2., if you're mixing a mouse workflow with a keyboard workflow, it's going to be clunky. Focusing the tree view with `ctrl-\` allows you to immediately hit `a` to add a new file. You can also use the arrows and enter key to browse and toggle collapsing.
